extensions/2.0/Vendor/EXT_meshopt_compression/README.md

```diff
@@ -0,0 +1,480 @@
+
+The encoded stream structure is as follows:
+
+- Header byte, which must be equal to `0xd1`
+- A sequence of index deltas, with encoding specified below
+- Tail block, which consists of 4 bytes that are reserved and should be set to 0
+
+Instead of simply encoding deltas vs the previous index, the decoder tracks *two* baseline index values, that start at 0. Each delta is specified in relation to one of these values and updates it so that the next delta that references the same baseline uses the encoded index value as a reference. This encoding is more efficient at handling some types of bimodal sequences where two independent monotonic sequences are spliced together, which can occur for some common cases of triangle strips or line lists.
+
+To specify the index delta, the varint-7 encoding scheme is used, which encodes an integer as one or more bytes, with the byte with the 0 most significant bit terminating the sequence:
+
+```
+0x7F => 0x7F
+0x81 0x04 => 0x201
+0xFF 0xA0 0x05 => 0x1FD005
+```
+
+When decoding the deltas, the 32-bit value is read using the varint encoding. The least significant bit of the value indicates one of the baseline values; the remaining bits specify a zigzag-encoded signed delta and can be decoded as follows:
+
+```
+uint32_t decode(uint32_t v) {
+	int32_t baseline = v & 1;
+	int32_t delta = (v & 2) != 0 ? ~(v >> 2) : (v >> 2);
+
+	last[baseline] += delta;
+	return last[baseline];
+}
+```
+
+It's up to the encoder to determine the optimal selection of the baseline for each index; this encoding scheme can be used to do basic delta encoding (with baseline bit always set to 0) as well as more complex bimodal encodings.
+
+Note that the zigzag-encoded delta must fit in a 31-bit integer; as such, deltas are limited to [-2^30..2^30-1].
+
+# Appendix B: Filters
+
+Filters are functions that transform each encoded attribute. For each filter, this document specifies the transformation used for decoding the data; it's up to the encoder to pick the parameters of the encoding for each element to balance quality and precision.
+
+For performance reasons the results of the decoding process are specified to one unit in last place (ULP) in terms of the decoded data, e.g. if a filter results in a 16-bit signed normalized integer, decoding may produce results within 1/32767 of specifed value.
+
+## Filter 1: octahedral
+
+Octahedral filter allows to encode unit length 3D vectors (normals/tangents) using octahedral encoding, which results in a more optimal quality vs precision tradeoff compared to storing raw components.
+
+The input to the filter is four 8-bit or 16-bit components, where the first two specify the X and Y components in octahedral encoding encoded as signed normalized K-bit integers, the third component explicitly encodes 1.0 as a K-bit integer, and the last component may contain arbitrary data (which is useful for tangents).

```

Comment:
To avoid any ambiguity, please
- mention whether K has any constraints;
- define "signed normalized K-bit integer" (e.g., two-complement with a sign in the MSB);
- add an example of "1.0 encoded as a K-bit integer" as only the first two components are specified as "signed normalized", mention that it's used to determine K, and what to do with invalid values in this component;
- define `INTN_MAX`.

Reply:
Thanks, I've tightened this description a bit and added language around the 1.0 encoding (both what is encoded and how it's used).